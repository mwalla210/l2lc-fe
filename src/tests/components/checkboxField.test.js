import React from 'react'
import renderer from 'react-test-renderer'
import CheckboxField from '../../components/checkboxField'

const defaultOptions = {
  id: 'id',
  disabled: false,
  value: false,
  onChange: jest.fn()
}

describe('CheckboxField', () => {
  it ('Renders with snapshot (disabled: false, value: false)', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <CheckboxField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: true, value: false)', () => {
    let options = Object.assign({}, defaultOptions)
    options.disabled = true
    const component = renderer.create(
      <CheckboxField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: true, value: true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.disabled = true
    options.value = true
    const component = renderer.create(
      <CheckboxField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: false, value: true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.value = true
    const component = renderer.create(
      <CheckboxField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
