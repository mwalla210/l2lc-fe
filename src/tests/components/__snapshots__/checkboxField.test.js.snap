// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField Renders with snapshot (disabled: false, value: false) 1`] = `
<input
  checked={false}
  className="form-check-input"
  disabled={false}
  id="id"
  onChange={[MockFunction]}
  style={
    Object {
      "margin": 5,
    }
  }
  type="checkbox"
/>
`;

exports[`CheckboxField Renders with snapshot (disabled: false, value: true) 1`] = `
<input
  checked={true}
  className="form-check-input"
  disabled={false}
  id="id"
  onChange={[MockFunction]}
  style={
    Object {
      "margin": 5,
    }
  }
  type="checkbox"
/>
`;

exports[`CheckboxField Renders with snapshot (disabled: true, value: false) 1`] = `
<input
  checked={false}
  className="form-check-input"
  disabled={true}
  id="id"
  onChange={[MockFunction]}
  style={
    Object {
      "margin": 5,
    }
  }
  type="checkbox"
/>
`;

exports[`CheckboxField Renders with snapshot (disabled: true, value: true) 1`] = `
<input
  checked={true}
  className="form-check-input"
  disabled={true}
  id="id"
  onChange={[MockFunction]}
  style={
    Object {
      "margin": 5,
    }
  }
  type="checkbox"
/>
`;
